crates/fj-window/src/run.rs

```diff
@@ -46,39 +46,43 @@ pub fn run(
     let mut draw_config = DrawConfig::default();
 
     let mut shape = None;
-    let mut camera = None;
+    let mut camera = Camera::new(&Default::default());
+    let mut camera_update_once = watcher.is_some();
 
     event_loop.run(move |event, _, control_flow| {
         trace!("Handling event: {:?}", event);
 
-        if let Some(new_shape) = watcher.receive(&mut status) {
-            match shape_processor.process(&new_shape) {
-                Ok(new_shape) => {
-                    renderer.update_geometry(
-                        (&new_shape.mesh).into(),
-                        (&new_shape.debug_info).into(),
-                        new_shape.aabb,
-                    );
+        if let Some(watcher) = &watcher {
+            if let Some(new_shape) = watcher.receive(&mut status) {
+                match shape_processor.process(&new_shape) {
+                    Ok(new_shape) => {
+                        renderer.update_geometry(
+                            (&new_shape.mesh).into(),
+                            (&new_shape.debug_info).into(),
+                            new_shape.aabb,
+                        );
 
-                    if camera.is_none() {
-                        camera = Some(Camera::new(&new_shape.aabb));
-                    }
+                        if camera_update_once {
+                            camera_update_once = false;
+                            camera = Camera::new(&new_shape.aabb);
+                        }
```

Comment:
I'm not sure I like `camera_update_once`. It seems like the kind of thing that could possibly turn into a bug, as things get shuffled around over time. But it *is* nicer, than having `if let Some(camera)` everywhere.

I think maybe `Camera` is no longer well-suited to this new situation. We could make the constructor parameter-less, just creating reasonable defaults there, and have some kind of internal flag that makes sure that `update_planes` does what `Camera::new` currently does when first called.

Then we could just create the camera using `Camera::new` above and do an unconditional `update_planes` here, instead of down there right before the rendering. But that would be work for a follow-up pull request (and if you're inclined to do it, I'd be happy to merge it).